app/forms/new_deputy_assignment_form.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+class NewDeputyAssignmentForm
+  class IdentityServiceError < StandardError; end
+  class IdentityNotFound < StandardError; end
+  class AlreadyAssigned < StandardError; end
+  class ErrorCreatingUser < StandardError; end
+
+  include ActiveModel::Model
+
+  attr_accessor :primary,
+                :deputy_webaccess_id
+
+  validates :primary,
+            presence: true
+
+  validates :deputy_webaccess_id,
+            presence: true,
+            format: {
+              with: /\A[a-z0-9]+\z/i,
+              allow_blank: true
+            }
+
+  def save
+    return false unless valid?
+
+    begin
+      deputy = find_or_initialize_deputy(webaccess_id: deputy_webaccess_id)
+      create_deputy_assignment(primary: primary, deputy: deputy)
+    rescue IdentityServiceError
+      errors.add(:base, :identity_service_error)
+    rescue IdentityNotFound
+      errors.add(:deputy_webaccess_id, :not_found)
+    rescue AlreadyAssigned
+      errors.add(:deputy_webaccess_id, :already_assigned)
+    rescue ErrorCreatingUser
+      errors.add(:base, :error_creating_user)
+    rescue StandardError
+      errors.add(:base, :unknown_error)
+    end
```

Comment:
Rescuing all the errors strikes me as a bit of a code smell. Could we do the same thing with nil checks, which are faster, and then add the errors as we go?

Reply:
This line exists to catch network errors coming from the PsuIdentity gem. Network timeouts, 4xx responses, etc, which could not be handled by nil checks. As best I could tell, that gem does not have a base class for its errors, and right now uses Faraday (which _does_ have a base class). However, that gem could easily _not_ use Faraday in future versions, so it would be foolish to catch Faraday error classes here. If PsuIdentity had its own base class for errors, that's what I'd use on this line

Reply:
https://github.com/psu-libraries/psu_identity/issues/12